auth/server/base/src/main/java/org/wildfly/security/auth/server/SecurityRealm.java

```diff
@@ -63,6 +64,11 @@ default RealmIdentity getRealmIdentity(Evidence evidence) throws RealmUnavailabl
         final Principal principal = evidence.getDecodedPrincipal();
         return principal == null ? RealmIdentity.NON_EXISTENT : getRealmIdentity(principal);
     }
+    default RealmIdentity getRealmIdentity(Evidence evidence, Function<Principal, Principal> principalTransformer) throws RealmUnavailableException {
```

Comment:
Would be good to add javadoc for this method.

Reply:
Done